### PR TITLE
Improve VCF parsing

### PR DESCRIPTION
**Pull request by jameshadfield**
This isn't complete yet, but if people wanted to take a look and see if they are happy with the direction this is going in that'd be great!

I still want to take a close look at how insertions are being encoded in the "sequences" output, and is treetime/augur is ok with that (i.e. `data['sequences'][<sample>][<pos>]` is more than one character). And ideally I'd sort out the writing of VCFs as well to both preserve the ploidy representation of alleles and also use the correct chromosome name!

### CODE REVIEW

**Comment by jameshadfield** [comment]
@rneher could you suggest someone to review it (or look at it yourself?) 🙏 

There are a number of bugfixes related to parsing haploid (or polyploid homozygous) VCFs and I'm now reasonably confident the parsing is correct. The encoding of insertions has the potential to double count the base before the insertion (see comments added in code here) but I don't know what treetime does with these; for the purposes of `augur ancestral`, which is what motivated this work, insertions are not considered.

I haven't touched the parsing of heterozygous genotypes.

I've fixed a number of bugs in the writing of VCFs, but I haven't thoroughly reviewed that code and the test VCF I've added here is pretty simple.

File: treetime/vcf_utils.py [hunk 1]
```diff
@@ -494,7 +576,7 @@ def handleDeletions(i, pi, pos, ref, delete, pattern):
         #Write it out - Increment positions by 1 so it's in VCF numbering
         #If no longer variable, and explained, don't write it out
         if printPos:
-            output = ["MTB_anc", str(pos), ".", refb, ",".join(uniques), ".", "PASS", ".", "GT"] + calls
+            output = [chrom_name, str(pos), ".", refb, ",".join(uniques), ".", "PASS", ".", "GT"] + calls
```
Comment (jameshadfield) [inline treetime/vcf_utils.py:579]: Running through `bcftools view` errors with
```
[W::vcf_parse] contig '1' is not defined in the header. (Quick workaround: index the file with tabix.)
[vcf.c:1265 bcf_write] Unchecked error (1), exiting.
```

This error's not new, but I should fix it here

**Comment by rneher** [comment]
this all looks good to me. happy to merge it in and make a release.

**Comment by jameshadfield** [comment]
> this all looks good to me. happy to merge it in and make a release.

Thanks! I'm just doing some final checks comparing `augur ancestral` & `augur translate` using FASTA vs VCF inputs on the Cholera dataset so let's hold off on this for a few days just in case there are any more little fixes I need to add.

File: treetime/vcf_utils.py
Comment (victorlin) [inline treetime/vcf_utils.py:1]: I understand that [PyVCF](https://github.com/jamescasbon/PyVCF) (or the more current [PyVCF3](https://github.com/dridk/PyVCF3)) is not used to replace/supplement `read_vcf` for [parsing efficiency reasons](https://github.com/neherlab/treetime/blob/9fdceaa5157fd0fd4bb141f45e1d742167e5d4c9/treetime/vcf_utils.py#L177-L180), but could it be useful for `write_vcf`?

Comment (jameshadfield) [reply treetime/vcf_utils.py:1]: Thanks partly to your suggestion to use cProfile, the VCF writing is now down to a handful of seconds [58ab78d](https://github.com/neherlab/treetime/pull/263/commits/58ab78dd59d77aa37a369f62d4c414d770e395f0) 🎉

**Comment by jameshadfield** [comment]
@rneher I'm happy with the state of this now and don't plan to do any more work here. There were a few more commits added since you last saw it. Once this is merged and released I'll update https://github.com/nextstrain/augur/pull/1355 to use the newest version.

**Pull request merged**